### DISCUSSION TITLE
there are missing modes on your integration

### DISCUSSION THREAD

**Discussion by titof2375** [omerfaruk-aran/esphome_samsung_hvac_bus#164] [General]
Hello, on my Samsung air conditioning unit ref AR07TXEAAWKNEU, I have the option of turning off the light and the beep on the cassette.

**Comment by omerfaruk-aran** [comment]
> Hello, on my Samsung air conditioning unit ref AR07TXEAAWKNEU, I have the option of turning off the light and the beep on the cassette.

Hi, before we proceed further, could you please check if you are able to control the beep sound and the light of your Samsung air conditioning unit via the SmartThings app? If you can, it would be helpful if you could enable logging on your ESP device and then try controlling the beep and light through the SmartThings app. After that, please send us the log files generated by the ESP device during this process. This will help us better understand how these features are managed and potentially integrate or control them via the ESP.

**Comment by titof2375** [comment]
[logs_clim-samsung_logs.txt](https://github.com/user-attachments/files/16686129/logs_clim-samsung_logs.txt)
Hello, yes absolutely, I control the bit and the light not the SmartThings app and here are the logs you told me to do.

**Reply by lanwin** [reply]
Did you have a separate smart things box or do you connected them to the indoor units directly?

**Reply by titof2375** [reply]
Hello, I am connected directly to my indoor units

**Reply by lanwin** [reply]
Then we have no chance to grab the message cause smarthings directly change the value without sending a message out to the bus.

**Comment by lanwin** [comment]
Currently we have no way to control that. Simply cause we did not know the messages to change that.

**Reply by titof2375** [reply]
That's precisely why I'm asking you because you are more capable than I found, the logs started when I modified the beeps and the light.  Plus I know the command in json that might help

**Reply by titof2375** [reply]
![Uploading Screenshot_20240821-090740.png…]()

**Reply by titof2375** [reply]
I have this code though
`[
            "mode/vs/0",
            {
               "x.com.samsung.da.options":[
                  "Light_Off"
               ]
            }
         ]
`

it's json, I don't know how to do it

**Reply by lanwin** [reply]
Where did you get that from? This did not came out of this codebase.

**Reply by titof2375** [reply]
This was found on this discussion group https://community.smartthings.com/

**Comment by titof2375** [comment]
Hello there is a discussion on the beep which in my opinion looks a lot like my request. here is the link https://github.com/lanwin/esphome_samsung_ac/issues
